src/edge-agent/Agent.Credentials.ts

```diff
@@ -61,52 +66,88 @@ export class AgentCredentials implements AgentCredentialsClass {
+    type: Type,
+    definition: PresentationDefinitionRequest<Type>,
+    from: DID,
+    to: DID
+  ) {
+    const attachmentFormat = type === CredentialType.JWT ?
+      AttachmentFormats.PRESENTATION_EXCHANGE_DEFINITIONS :
+      AttachmentFormats.ANONCREDS_PROOF_REQUEST;
+
+    return new RequestPresentation(
+      {
+        proofTypes: [],
+      },
+      [
+        AttachmentDescriptor.build(
+          definition,
+          uuid(),
+          'application/json',
+          undefined,
+          attachmentFormat
+        )
+      ],
+      from,
+      to,
+      uuid()
+    );
+  }
+
   async initiatePresentationRequest(
     type: CredentialType,
     toDID: DID,
-    claims: PresentationClaims): Promise<RequestPresentation> {
-
+    claims: PresentationClaims<CredentialType>
+  ): Promise<RequestPresentation> {
     const didDocument = await this.castor.resolveDID(toDID.toString());
     const newPeerDID = await this.agentDIDHigherFunctions.createNewPeerDID(
       didDocument.services,
       true
     );
 
-    const options = new PresentationOptions({
-      jwt: {
-        jwtAlg: ['ES256K']
-      },
-      challenge: "Sign this text " + uuid(),
-      domain: 'N/A'
-    });
-
-    const presentationDefinitionRequest = await this.pollux.createPresentationDefinitionRequest(
-      type,
-      claims,
-      options
-    );
-
-    const attachment = AttachmentDescriptor.build(
-      presentationDefinitionRequest,
-      uuid(),
-      'application/json',
-      undefined,
-      AttachmentFormats.PRESENTATION_EXCHANGE_DEFINITIONS
-    )
-
-    const requestPresentationBody: RequestPresentationBody = {
-      proofTypes: [],
+    if (type === CredentialType.AnonCreds) {
+      if (!validatePresentationClaims(claims, CredentialType.AnonCreds)) {
+        throw new PolluxError.InvalidPresentationDefinitionError("Anoncreds Claims are invalid")
+      }
+      const presentationDefinitionRequest = await this.pollux.createPresentationDefinitionRequest(
+        type,
+        claims,
+        new PresentationOptions({}, CredentialType.AnonCreds)
+      )
+      return this.createPresentationDefinitionRequest<CredentialType.AnonCreds>(
+        type,
+        presentationDefinitionRequest,
+        newPeerDID,
+        toDID
+      )
     }
 
-    const presentationRequest = new RequestPresentation(
-      requestPresentationBody,
-      [attachment],
-      newPeerDID,
-      toDID,
-      uuid()
-    );
+    if (type === CredentialType.JWT) {
+      if (!validatePresentationClaims(claims, CredentialType.JWT)) {
+        throw new PolluxError.InvalidPresentationDefinitionError("JWT Claims are invalid")
+      }
+      const presentationDefinitionRequest = await this.pollux.createPresentationDefinitionRequest<CredentialType.JWT>(
+        type,
+        claims,
+        new PresentationOptions({
+          jwt: {
+            jwtAlg: ['ES256K']
+          },
+          challenge: "Sign this text " + uuid(),
```

Comment:
Challenge should be something that is rotating over time so that we never request the user to sign the same message. 
For now we don't let the user define the challenge, but also the holder does not choose what he wants to sign, if u know what i mean, this happens in background and the signing + verifying process is automated.